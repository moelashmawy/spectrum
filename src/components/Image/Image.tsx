import React from "react";
import styled from "styled-components";

const StyledImage = styled.img<{ $isAscending?: boolean }>`
  transform: ${(props) =>
    props.$isAscending ? "rotate(0deg)" : "rotate(90deg)"};
  transition: transform 0.3s ease-in-out;
`;

interface ImageProps {
  isAscending: boolean;
  image: string;
  width?: number;
  height?: number;
}

const Image: React.FC<ImageProps> = ({
  isAscending,
  image,
  width = 60,
  height = 60,
}) => {
  return (
    <StyledImage
      height={height}
      width={width}
      src={image}
      $isAscending={isAscending}
    />
  );
};

export default Image;
