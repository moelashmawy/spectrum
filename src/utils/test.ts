import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ReactElement } from "react";

export const createSnapshot = (ComponentToTest: ReactElement) => {
  const {
    container: { firstChild: actual },
  } = render(ComponentToTest);
  expect(actual).toMatchSnapshot();
};
