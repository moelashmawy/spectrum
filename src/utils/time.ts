export const getCurrentTime = () => {
  const d = new Date();
  return d.toLocaleTimeString();
};
