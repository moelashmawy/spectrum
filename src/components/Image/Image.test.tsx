import Image from "./Image";
import { createSnapshot } from "../../utils/test";

describe("Image", () => {
  test("Image snapshot when isAscending", () => {
    createSnapshot(
      <Image isAscending image="mock_img.jpg" width={60} height={60} />,
    );
  });

  test("Image snapshot when not isAscending", () => {
    createSnapshot(
      <Image isAscending={false} image="mock_img.jpg" width={60} height={60} />,
    );
  });
});
