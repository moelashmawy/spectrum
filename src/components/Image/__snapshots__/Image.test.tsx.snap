// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Image snapshot when isAscending 1`] = `
<img
  class="sc-bdfCDU eEOjSd"
  height="60"
  src="mock_img.jpg"
  width="60"
/>
`;

exports[`Image Image snapshot when not isAscending 1`] = `
<img
  class="sc-bdfCDU bRkxDg"
  height="60"
  src="mock_img.jpg"
  width="60"
/>
`;
